# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/android/Platform.cpp
# Commit: 061c08a40229
# Full Hash: 061c08a40229ab34058e5962a0fe6b0d5250b0a3
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2023-12-07 09:47:19
# Description:
#   Bug 1868092 - Don't call DoPivot on a null leaf in caret move. r=morgan
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195599
# ==============================================================================

diff -r 2805138dc4b4 -r 061c08a40229 accessible/android/Platform.cpp
--- a/accessible/android/Platform.cpp	Wed Dec 06 21:57:33 2023 +0000
+++ b/accessible/android/Platform.cpp	Wed Dec 06 22:06:07 2023 +0000
@@ -163,10 +163,12 @@
     // This is used mostly for find in page.
     Accessible* leaf = TextLeafPoint::GetCaret(aTarget).ActualizeCaret().mAcc;
     MOZ_ASSERT(leaf);
-    if (Accessible* result = AccessibleWrap::DoPivot(
-            leaf, java::SessionAccessibility::HTML_GRANULARITY_DEFAULT, true,
-            true)) {
-      sessionAcc->SendAccessibilityFocusedEvent(result);
+    if (leaf) {
+      if (Accessible* result = AccessibleWrap::DoPivot(
+              leaf, java::SessionAccessibility::HTML_GRANULARITY_DEFAULT, true,
+              true)) {
+        sessionAcc->SendAccessibilityFocusedEvent(result);
+      }
     }
   }
 
